maze.pl

```diff
@@ -135,10 +142,13 @@
 %
 % Ensure that the tests pass with no warnings.
 %
+path_to(Start,End, Route):- connected(Start, End, [],R), rev(R, [], Route).
+connected(X,X,T,[X|T).
+connected(Place, End, T, R):- corridor(Place, Next), mem(Next,T), connected(Next,End,[Place|t],R).
```

Comment:
It would have been a little less confusing if the path_to helper was not named connected, as we already have that as a preddicate.

In any case, there is a syntax error here (the unbalance braces).  Also, I think the atom 't' should have been the variable 'T'.  You also did not use negate the membership test.